vaadin-context-menu-flow-parent/vaadin-context-menu-flow-integration-tests/src/main/java/com/vaadin/flow/component/contextmenu/it/MenuItemDisableOnClickView.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2000-2024 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.contextmenu.it;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.vaadin.flow.component.contextmenu.ContextMenu;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.router.Route;
+
+@Route("vaadin-context-menu/disable-on-click")
+public class MenuItemDisableOnClickView extends Div {
+
+    static String getEnabledStateChangeMessage(String itemName,
+            boolean newEnabledState, int numberOfClicks) {
+        return "Item " + itemName
+                + " was clicked and enabled state was changed to "
+                + newEnabledState + " receiving " + numberOfClicks + " clicks";
+    }
+
+    private final ContextMenu contextMenu;
+
+    public MenuItemDisableOnClickView() {
+        contextMenu = new ContextMenu();
+        addItemWithDisableOnClick();
+        addItemWithDisableOnClickThatEnablesInSameRoundTrip();
+        addItemWithDisableAndHideOnClick();
+        addItemWithDisableOnClickAndPointerEventsAuto();
+        addDetachReattachItem();
+        contextMenu.getItems().forEach(item -> item.setKeepOpen(true));
+
+        var toggleKeepOpen = new NativeButton("Toggle keep open",
+                click -> contextMenu.getItems()
+                        .forEach(item -> item.setKeepOpen(!item.isKeepOpen())));
+        toggleKeepOpen.setId("toggle-keep-open");
+        add(toggleKeepOpen);
+
+        var target = new Div("Target");
+        target.setId("target-div");
+        contextMenu.setTarget(target);
+        add(target);
+    }
+
+    private void addItemWithDisableOnClick() {
+        var disableOnClickMenuItem = contextMenu.addItem("Disabled on click",
+                event -> {
+                    // Triggering an action that can be started only once
+                });
+        disableOnClickMenuItem.setDisableOnClick(true);
+
+        var temporarilyDisabledMenuItem = contextMenu
```

Comment:
This menu item and the "enable in same roundtrip" one seem to have the same setup. Both disable and reenable the item in the same roundtrip. I'm not sure if I get the point of the temporarily disabled item test.

Reply:
Removed the "temporarily disabled item" and also removed that test.